src/IR/IR.hs

```diff
@@ -252,3 +255,97 @@ wellFormed (Prim p es _) = wfPrim p es && all wellFormed es
   wfPrim (PrimOp PrimLt    ) [_, _]    = True
   wfPrim (PrimOp PrimLe    ) [_, _]    = True
   wfPrim _                   _         = False
+
+
+{-
+how to use:
+1) stack ghci
+2) import IR.IR
+3) samples:
+show (LitIntegral 3)
+show New
+Questions:
+why does `show (Lit "a" Integer)` not work? -- b/c types not defined, just use () as a type
+why does `show (Var (VarId (Identifier "foo")) LitIntegral)` not work?
+
+show (Var (VarId (Identifier "foo")) ()))
+
+-}
+instance (Pretty t) => Show (Expr t) where
+  show p = show $ pretty p
+
+instance Show Primitive where
+  show p = show $ pretty p
+
+instance Show Literal where
+    show p = show $ pretty p
+
+instance (Pretty t) => Show (Program t) where
+    show p = show $ pretty p
+
+instance Pretty Literal where
+    pretty (LitIntegral i) = pretty (show i)
+    pretty (LitBool b) = pretty (show b)
+
+instance Pretty Primitive where
+  pretty New = pretty "new"
+  pretty Dup = pretty "dup"
+  pretty Drop = pretty "drop"
+  pretty Reuse = pretty "reuse"
+  pretty Deref = pretty "deref"
+  pretty Assign = pretty "="
```

Comment:
Assign should denoted by an infix `<-`; something like `Assign x y` should be `x <- y`.

You may want to consider not having `Primitive` as an instance of `Show`, and instead matching on each enum explicitly in `pretty (Prim ..)` in `instance Pretty Expr t where ...`.